exchange/exchange.go

```diff
@@ -484,18 +499,17 @@ func applyCategoryMapping(ctx context.Context, requestExt openrtb_ext.ExtRequest
 
 	}
 	if len(seatBidsToRemove) > 0 {
-		if len(seatBidsToRemove) == len(seatBids) {
-			//delete all seat bids
-			seatBids = nil
-		} else {
-			for _, seatBidInd := range seatBidsToRemove {
-				delete(seatBids, seatBidInd)
-			}
-
+		for _, seatBidInd := range seatBidsToRemove {
+			seatBids[seatBidInd].bids = nil
 		}
 	}
 
-	return res, seatBids, nil
+	return res, seatBids, rejections, nil
+}
+
+func updateRejections(rejections []string, bidID string, reason string) []string {
+	message := fmt.Sprintf("bid rejected [bid ID: %s] reason: %s", bidID, reason)
+	return append(rejections, message)
```

Comment:
Does this checkout ok? Appending to the rejections would alter the slice, so wouldn't this need to be a reference?

Reply:
From what I've seen it's a pretty common pattern with slices to pass by value and return a modified slice rather than passing by reference, the append function itself is a good example of that. That said I do agree with you that        it's more confusing that way, but it's idoimatic. I guess what I'm saying is I don't really have an opinion after all. :)

Reply:
A slice is basically 3 bits of data. A pointer to some memory, the length of the slice that extends beyond that pointer, and the max size that the slice can expand to given the current memory allocation. So effectively it is a pointer as far as the data contained. The resulting slice is returned, because if we overrun the max size, we will have to allocate a new chunk of memory for the larger slice, which will have no relation to the current chunk of memory. Thus we push up the replaced slice back to the caller so it will have the correct data there. This is how the built in `append()` function works.